frontend/src/components/Forms/CoffeeChatsForm/CoffeeChatsForm.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from 'react';
+import { Form, Dropdown } from 'semantic-ui-react';
+import { Emitters } from '../../../utils';
+import { useSelf } from '../../Common/FirestoreDataProvider';
+import { MembersAPI } from '../../../API/MembersAPI';
+import CoffeeChatsDashboard from './CoffeeChatsDashboard';
+import styles from './CoffeeChatsForm.module.css';
+import CoffeeChatAPI from '../../../API/CoffeeChatAPI';
+
+const CoffeeChatsForm: React.FC = () => {
+  const userInfo = useSelf()!;
+  const [member, setMember] = useState<IdolMember | undefined>(undefined);
+  const [category, setCategory] = useState('');
+  const [membersList, setMembersList] = useState<IdolMember[]>([]);
+  const [approvedChats, setApprovedChats] = useState<CoffeeChat[]>([]);
+  const [pendingChats, setPendingChats] = useState<CoffeeChat[]>([]);
+  const [rejectedChats, setRejectedChats] = useState<CoffeeChat[]>([]);
+  const [isChatLoading, setIsChatLoading] = useState<boolean>(true);
+  const [slackLink, setSlackLink] = useState<string>('');
+
+  useEffect(() => {
+    MembersAPI.getAllMembers().then((members) => setMembersList(members));
+    CoffeeChatAPI.getCoffeeChatsByUser(userInfo).then((coffeeChat) => {
+      setApprovedChats(coffeeChat.filter((chat) => chat.status === 'approved'));
+      setPendingChats(coffeeChat.filter((chat) => chat.status === 'pending'));
+      setRejectedChats(coffeeChat.filter((chat) => chat.status === 'rejected'));
+      setIsChatLoading(false);
+    });
+  }, [userInfo]);
+
+  const coffeeChatExists = async (): Promise<boolean> => {
+    const chats = await CoffeeChatAPI.getCoffeeChatsByUser(userInfo);
+    return chats.some((chat) => chat.otherMember.netid === member?.netid);
+  };
+
+  const submitCoffeeChat = async () => {
+    if (!member) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Member Selected',
+        contentMsg: 'Please select a team member!'
+      });
+    } else if (!slackLink) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Image Entered',
+        contentMsg: 'Please enter an image link!'
+      });
+    } else if (!category) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Category Entered',
+        contentMsg: 'Please enter a category!'
+      });
+    } else if (member.netid === userInfo.netid) {
+      Emitters.generalError.emit({
+        headerMsg: 'Cannot Coffee Chat Yourself',
+        contentMsg: 'Please submit a coffee chat with another member!'
+      });
+    } else if (await coffeeChatExists()) {
+      Emitters.generalError.emit({
+        headerMsg: 'Coffee Chat Exists',
+        contentMsg: 'Please submit a coffee chat with a new member!'
+      });
+    } else {
+      const newCoffeeChat: CoffeeChat = {
+        uuid: '',
+        submitter: userInfo,
+        otherMember: member,
+        slackLink,
+        category,
+        status: 'pending' as Status,
+        date: new Date().getTime()
```

Comment:
Personal opinion here is not to trust the frontend on the date. I think it's fine that you're putting something here for the value, but maybe overwrite it on the backend with the time there?